azurelinuxagent/ga/long_polling.py

```diff
@@ -0,0 +1,82 @@
+import datetime
+import threading
+import time
+
+from azurelinuxagent.common.interfaces import ThreadHandlerInterface
+from azurelinuxagent.common import logger
+from azurelinuxagent.common.future import httpclient, ustr
+
+
+def get_longpolling_handler():
+    return LongPollingHandler()
+
+
+class LongPollingHandler(ThreadHandlerInterface):
+    _THREAD_NAME = "LongPollingHandler"
+
+    @staticmethod
+    def get_thread_name():
+        return LongPollingHandler._THREAD_NAME
+
+    def __init__(self):
+        self.longpolling_thread = None
+        self.should_run = True
+
+    def run(self):
+        self.start()
+
+    def stop(self):
+        self.should_run = False
+        if self.is_alive():
+            self.join()
+
+    def join(self):
+        self.longpolling_thread.join()
+
+    def stopped(self):
+        return not self.should_run
+
+    def keep_alive(self):
+        return False
+
+    def is_alive(self):
+        return self.longpolling_thread is not None and self.longpolling_thread.is_alive()
+
+    def start(self):
+        self.longpolling_thread = threading.Thread(target=self.daemon)
+        self.longpolling_thread.setDaemon(True)
+        self.longpolling_thread.setName(self.get_thread_name())
+        self.longpolling_thread.start()
+
+    def daemon(self):
+        new_request = True
+        while new_request:
+            try:
+                logger.info("[HTTP Request(longPolling)] Timestamp:{0}".format(datetime.datetime.utcnow()))
+
+                timeout = 60
+                conn = httpclient.HTTPConnection("168.63.129.16", 32526, timeout=10)
```

Comment:
Why have two timeouts? why not just set this timeout to 60?

Reply:
Timeout here just connection timeout is the number of seconds requests will wait for your client to establish a connection but doesn't represent how long client should wait for response. Unfortunately, this library doesn't support unlike requests library https://stackoverflow.com/questions/21965484/timeout-for-python-requests-get-entire-response?msclkid=75b3309fc72a11eca5ad6e51d51368ff

So, I had to come up with second timeout just for Read timeout. https://stackoverflow.com/questions/49704708/what-is-a-connection-timeout-during-a-http-request?msclkid=43fb8e06c72b11ecbc1c813c14fa0c6e